SimulationAgent/SimulationRunner.cs

```diff
@@ -156,6 +161,17 @@ public async Task StartAsync(Simulation simulation)
             // of threads pending on the lock statement
             if (this.starting || this.stopping || this.running) return;
 
+            // Loop through all the device models used in the simulation
+            var models = (from model in simulation.DeviceModels where model.Count > 0 select model).ToList();
+
+            // Calculate the total number of devices
+            // the active hub, e.g. in case the user provided a custom connection string.
+            var total = models.Sum(model => model.Count);
+
+            // Send a request to update vmss auto scale settings to create vm instances
+            var vmCount = (int)Math.Ceiling((double)total / this.clusteringConfig.MaxDevicesPerNode);
+            await this.azureManagementAdapter.CreateOrUpdateVmssAutoscaleSettingsAsync(vmCount);
```

Comment:
The code under the simulation agent should not have to deal with infrastructure or similar "setup" tasks. Please move the cluster management code to the Partitioning Agent, in particular limit it to the Master node